### PR TITLE
CI - Automatically run LightHouse (on PR commits)

### LINKED ISSUE #78: Automated LightHouse reporting as PR comment (Github actions CI)
# Goal

Run a lighthouse report when a commit is pushed, which eventually displays the lighthouse results as a PR comment.

https://github.com/foo-software/lighthouse-check-action/issues/24

This could run in parallel with E2E (or afterwards, I'm not sure if parallel github actions are possible)

### PR DESCRIPTION
**Pull request by Demmonius**
We want to create Lighthouse reports for each commit.
It should:

- Run in parallel with E2E tests
- Display a report
- Upload artifacts
- Make the workflow fail if the score is too low (@Vadorequest I'll need a minimal "critical" score to set :D )

I don't think we should have another PR comment about the Lighthouse reports, another spam. It should be enough to have a job which fail.

### CODE REVIEW

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 563e8bc4bf7b7d290375472a41e513edd9bafa35 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-nd3pgivvk.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 1567b8d55cb54726476cff1ba2e84e9ad4f58cce successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-d23b24bvz.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit d1bb0bb02e4633ee3a9abcc847ecd7d83a8c7297 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-ni5sanhof.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 16d273c9b1295151c388217e2ceb300eccbbda97 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-qerlzhg1n.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 1665312fea2fc7fed1ca176190232e9e0e0eb23d successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-q8pbmfl7f.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit d88f6219955ad17a1e2c587565167cfcce1beb57 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-fb9mjz2ap.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-lighthouse-inte.now.sh

**Comment by Demmonius** [comment]
@Vadorequest as you can see on this run: https://github.com/UnlyEd/next-right-now/pull/103/checks?check_run_id=757751879

If we want lighthouse to comment PR with the report, we need to change the trigger.
Instead of starting actions on a "push", we should restrict it for every commit IN a PR.

What do you think about it? Tell me if we:
- [x] want lighthouse to send message and change the trigger method
- [ ] don't care about lighthouse reports comment

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **FAILED**
Download artifacts (screenshots + videos) from [`checks`](https://github.com/UnlyEd/next-right-now/pull/103/checks) section

**Comment by Vadorequest** [comment]
> Make the workflow fail if the score is too low (@Vadorequest I'll need a minimal "critical" score to set :D )

We don't need that, LightHouse shouldn't make deployments fail due to low score, it's not the point. Warning the dev is enough IMO.

**Comment by Demmonius** [comment]
Lighthouse report is created after the deployment, by using the vercel url :P
It's not blocking at all. 
image

If the score is too low, it will prompt a "warning". Nothing more. We can also imagine, comment the PR only if score is too low :)

**Comment by Vadorequest** [comment]
So, a comment is necessary, something like what's printed in the GHA would be great.

But, why do you need to trigger it for each commit? Why not the latest commit being pushed? It'll create quite a lot of noise otherwise I'm afraid.

```
+--------------------------------------------------------------------------+
|                             Lighthouse Audit                             |
+--------------------------------------------------------------------------+

URL: https://nrn-v2-mst-aptd-at-lcz-sty-c1-8y7chl1to.now.sh
Local Report: /tmp/lighthouse-artifacts/lighthouse-report-1591792167426.html


+---------------+----------------+-------------+---------------------+-----+
| Accessibility | Best Practices | Performance | Progressive Web App | SEO |
|---------------|----------------|-------------|---------------------|-----|
| 100           | 93             | 78          | 74                  | 89  |
+---------------+----------------+-------------+---------------------+-----+
```

**Comment by Demmonius** [comment]
> Lighthouse report is created after the deployment, by using the vercel url :P
> It's not blocking at all.
> image
> 
> If the score is too low, it will prompt a "warning". Nothing more. We can also imagine, comment the PR only if score is too low :)

@Vadorequest up :D

**Comment by Vadorequest** [comment]
Let's try to make a comment for each commit first, and we'll see how it goes.

Anything below 30 of score should be looked into, so I supposed that can become the threshold :)

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit a3b8958c26a6446cddc9993ca83787364388ce63 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-5q0t60c21.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-ezrzfdk5k.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Demmonius** [comment]
>     
> 
> https://nrn-v2-mst-aptd-at-lcz-sty-c1-5q0t60c21.now.sh

As you can see @Vadorequest it looks nice !
The "checker" failed because performance score is lower than 30. What do you think about it? Is it usefull? Up to you :)

**Comment by Demmonius** [comment]
I think it's possible to do it by our own and only if the score is lower than 30. It would be just spam if we comment all the time...

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 1524fcc5935b0fa00069f1f4a5437f231cae5ea9 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-prcwy1n8m.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-prcwy1n8m.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit b6a930506ba1741d16307c1bfe6de4d91231a20a successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-n70qqd903.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
I prefer to keep it with one comment by push, it's what's the most useful to detect a regression in quality (SEO, good practices) and quickly locate it.

It may look like spam, but it's no different from the existing behaviour with E2E and deployments.

File: .github/workflows/deploy-zeit-staging.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 name: Deploy to Zeit (staging)
 
 on:
-  push:
+  pull_request:
```
Comment (Vadorequest) [inline .github/workflows/deploy-zeit-staging.yml]: Explain what that does, wasn't obvious to me

> Triggers on each commit associated with a pull request

File: .github/workflows/deploy-zeit-staging.yml [hunk 2]
```diff
@@ -147,35 +139,83 @@ jobs:
           name: videos
           path: cypress/videos/
 
-      # We need to find the PR id. Will be used later to comment on that PR.
```
Comment (Vadorequest) [inline .github/workflows/deploy-zeit-staging.yml:150]: why was that removed? 🤔

File: .github/workflows/deploy-zeit-staging.yml [hunk 3]
```diff
@@ -147,35 +139,83 @@ jobs:
...
+        #  - and then we remove the `"` character to pre-format url
+        # We need to set env the url for next step, formatted as `https://${url provided by API}`
+        run: |
+          apt update -y >/dev/null && apt install -y jq >/dev/null
+          ZEIT_DEPLOYMENT=`curl -H 'Accept: application/json' -H 'Content-Type: application/json' -H 'Authorization: Bearer ${{ secrets.ZEIT_TOKEN }}' https://api.zeit.co/v5/now/deployments?teamId=$(cat now.json | jq -r '.scope') | jq '.deployments [0].url' | tr -d \"`
+          echo "::set-env name=ZEIT_DEPLOYMENT_URL::https://$ZEIT_DEPLOYMENT"
+        env:
+          ZEIT_TOKEN: ${{ secrets.ZEIT_TOKEN }} # Passing github's secret to the worker
+      # In order to store reports and then upload it, we need to create the folder before any tests
+      - name: Create temporary fodler for artifacts storage
```
Comment (Vadorequest) [inline .github/workflows/deploy-zeit-staging.yml]: folder

File: .github/workflows/deploy-zeit-staging.yml [hunk 4]
```diff
@@ -147,35 +139,83 @@ jobs:
...
+
+      # Using a pre-build action to make the action fail if your score is too low. It can be really interesting to track a low score on a commit
+      # You can remove this action IF you don't want lighthouse to be a blocking point in your CI
+      # This default values so you need to change them by your own criteria
+      # Official documentation: https://github.com/foo-software/lighthouse-check-status-action
+      - name: Handle Lighthouse Check results
+        uses: foo-software/lighthouse-check-status-action@v1.0.1
+        with:
+          lighthouseCheckResults: ${{ steps.lighthouseCheck.outputs.lighthouseCheckResults }}
+          minAccessibilityScore: "30"
```
Comment (Vadorequest) [inline .github/workflows/deploy-zeit-staging.yml]: Change to 50 everywhere, except for minPerformanceScore use 30

**Review by Vadorequest — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-zeit-staging.yml (hunk 1 above)
Comment (Demmonius) [reply .github/workflows/deploy-zeit-staging.yml]: **push** means every commits. Our actions were run on each commit, even if no PR was open. That's the only difference.
**pull_request** means every commit IN a pull request :)

File: .github/workflows/deploy-zeit-staging.yml (hunk 2 above)
Comment (Demmonius) [reply .github/workflows/deploy-zeit-staging.yml:150]: Because now, we trigger the actions on a "pull_request" and not a "push". By this way, it's so much easier to find the PR id, we just have to use the variable `${{ github.event.number }}`

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 7ad2a61174e3c664419a5219f38c204c0ea793ba successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-ezrzfdk5k.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

File: .github/workflows/deploy-zeit-staging.yml (hunk 1 above)
Comment (Vadorequest) [reply .github/workflows/deploy-zeit-staging.yml]: Yes, now let's write it down to avoid further confusion :)

File: .github/workflows/deploy-zeit-staging.yml (hunk 2 above)
Comment (Vadorequest) [reply .github/workflows/deploy-zeit-staging.yml:150]: Great to know!

**Comment by Vadorequest** [comment]
@Demmonius Once it's cleaned for the staging file, could you update the production file as well?

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

File: .github/workflows/deploy-zeit-staging.yml (hunk 1 above)
Comment (Demmonius) [reply .github/workflows/deploy-zeit-staging.yml]: :'(

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 991d85c169e3387789a9a54d1b77840a1dd7d594 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-5o9wai0ar.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-5o9wai0ar.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 0f1649f379d620f7ada77d4b263e086e7a00656e successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-ihq212z7v.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-ihq212z7v.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 1360413bc8a55a65f846bd20331273f7e96a0142 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-a921way3x.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-a921way3x.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 1744043737cba00036f8079c5d4196686afb2b31 successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-2ecaveg2f.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-2ecaveg2f.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**

**Comment by Vadorequest** [comment]
[GitHub Actions]
Deployment **SUCCESS**
Commit 8fb24ac6a20de01b664bf0f2d5f0bd70be8b5aed successfully deployed to https://nrn-v2-mst-aptd-at-lcz-sty-c1-9o3qpj4oi.now.sh
Deployment aliased as https://nrn-v2-mst-aptd-at-lcz-sty-c1-merge.now.sh

**Review by Vadorequest — COMMENTED** [review]
https://nrn-v2-mst-aptd-at-lcz-sty-c1-9o3qpj4oi.now.sh

**Comment by Vadorequest** [comment]
[GitHub Actions]
E2E tests **SUCCESS**
